test/files/run/stream-gc.scala

```diff
@@ -8,4 +8,5 @@ object Test extends App {
   Stream.tabulate(100)(_ => new Array[AnyRef](10000)).collectFirst { case x if false => x }
   Stream.tabulate(100)(_ => new Array[AnyRef](10000)).collectFirst { case x if false => x }
   Stream.tabulate(100)(_ => new Array[AnyRef](10000)).collectFirst { case x if false => x }
+  Stream.tabulate(100)(_ => new Array[AnyRef](10000)).iterator.foreach(_ => ())
```

Comment:
did this test fail before the change?

Just making sure we have a test to catch any regression.

Reply:
yes, fails without the change. OOME.